utils/src/ooinstall/cli_installer.py

```diff
@@ -576,29 +594,32 @@ def get_missing_info_from_user(oo_cfg):
     confirm_continue(message)
     click.clear()
 
-    if oo_cfg.settings.get('ansible_ssh_user', '') == '':
-        oo_cfg.settings['ansible_ssh_user'] = get_ansible_ssh_user()
+    if not oo_cfg.deployment.variables.get('ansible_ssh_user', ''):
+        oo_cfg.deployment.variables['ansible_ssh_user'] = get_ansible_ssh_user()
         click.clear()
 
-    if oo_cfg.settings.get('variant', '') == '':
+    if not oo_cfg.deployment.variables.get('variant', ''):
         variant, version = get_variant_and_version()
-        oo_cfg.settings['variant'] = variant.name
-        oo_cfg.settings['variant_version'] = version.name
+        oo_cfg.deployment.variables['variant'] = variant.name
+        oo_cfg.deployment.variables['variant_version'] = version.name
         click.clear()
 
-    if not oo_cfg.hosts:
-        oo_cfg.hosts = collect_hosts(oo_cfg)
+    if not oo_cfg.deployment.hosts:
+        oo_cfg.deployment.hosts, roles = collect_hosts(oo_cfg)
+
+        for role in roles:
+            oo_cfg.deployment.roles.append(Role(name=role))
         click.clear()
 
-    if not oo_cfg.settings.get('master_routingconfig_subdomain', None):
-        oo_cfg.settings['master_routingconfig_subdomain'] = get_master_routingconfig_subdomain()
+    if not oo_cfg.deployment.variables.get('master_routingconfig_subdomain', None):
+        oo_cfg.deployment.variables['master_routingconfig_subdomain'] = get_master_routingconfig_subdomain()
```

Comment:
master_routingconfig_subdomain should be under the master role rather than the deployment.

Also, there should also be a short name mapping for it.